marklogic-unit-test-client/src/main/java/com/marklogic/test/unit/JaxpServiceResponseUnmarshaller.java

```diff
@@ -67,10 +68,14 @@ public TestSuiteResult parseTestSuiteResult(String xml) {
 			NodeList resultNodes = testNode.getChildNodes();
 			String failureXml = null;
 			for (int j = 0; j < resultNodes.getLength(); j++) {
-				Element resultNode = (Element) resultNodes.item(j);
-				if ("fail".equals(resultNode.getAttribute("type"))) {
-					failureXml = toXml(resultNode);
-					break;
+				if (resultNodes.item(j).getNodeType() == ELEMENT_NODE) { // process only element node, ignore the rest
+					Element resultNode = (Element) resultNodes.item(j);
+					if ("fail".equals(resultNode.getAttribute("type"))) {
+						failureXml = toXml(resultNode);
+						break;						
+					}
+				} else {
+					System.out.println("Ignoring Node Type [" + resultNodes.item(j).getNodeType() + "]");
```

Comment:
Libraries should almost never write something to sysout, as it can become very noisy for the user and can't be controlled. I don't think it's necessary to log anything here - if this were using a logger, I'd say it should be logged at the debug level, meaning it's rarely going to be helpful.

Reply:
Added logger with debug message